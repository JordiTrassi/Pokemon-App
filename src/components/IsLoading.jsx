import { CircularProgress, Typography, Box } from '@mui/material';

export const IsLoading = () => {
  
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        mt: 15,
      }}>
      <Typography sx={{fontSize: 30}}>Is Loading ...</Typography>
      <CircularProgress sx={{mt: 3}} />
    </Box>
  );
}
