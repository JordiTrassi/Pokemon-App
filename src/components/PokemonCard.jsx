import { Link } from 'react-router-dom';
import { Box, Card, CardContent, CardMedia, IconButton, Tooltip, Typography } from '@mui/material';
import { AddCircle } from '@mui/icons-material';




export const PokemonCard = ({name, url}) => {

  
  return (

    <Card
      className='animate__animated animate__fadeIn'
      sx={{
        display: 'flex',
        mb: '20px',
        justifyContent: 'space-between',
        boxShadow: 6,
        borderRadius: 5,
      }}
      >
      <Box sx={{ display: 'flex', flexDirection: 'column' }} >
        <CardContent sx={{ flex: '1 0 auto' }}>
          <Typography component="div" variant="h4">
            {
              name.toUpperCase()
            }
          </Typography>
        </CardContent>
              
        <Box sx={{ display: 'flex', alignItems: 'center', pl: 1, pb: 1 }}>
          <Link to={`${name}`} >
            <Tooltip
              title="Pokemon Info"
              placement="right"
              arrow
            >
              <IconButton>
                <AddCircle fontSize='large'/> 
              </IconButton>              
            </Tooltip>
          </Link>
               
        </Box>
      </Box>
      <CardMedia
        component="img"
        image={url}
        alt={name}
        sx={{ height: 200, width: 'auto' }}     
      />
    </Card>
  )
}


