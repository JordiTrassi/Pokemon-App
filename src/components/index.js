export * from './IsLoading';
export * from './NavBar';
export * from './PokemonCard';
