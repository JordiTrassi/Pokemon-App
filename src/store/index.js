export * from './store';
export * from './thunks';
export * from './slices/pokemonSlice';