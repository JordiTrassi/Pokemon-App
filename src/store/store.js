import { configureStore } from '@reduxjs/toolkit';
import { pokemonSlice } from './slices/pokemonSlice';

export const store = configureStore({
    reducer: {
      pokemonStore: pokemonSlice.reducer,
  },
})