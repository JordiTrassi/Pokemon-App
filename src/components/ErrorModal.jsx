import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Box, Typography, Modal } from '@mui/material';
import { Error } from '@mui/icons-material';
import { resetNoApiResults } from '../store';


const style = {
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: 400,
  bgcolor: 'background.paper',
  border: '2px solid #000',
  boxShadow: 24,
  p: 4,
};

export const ErrorModal = () => {

    const dispatch = useDispatch();
    const [open, setOpen] = useState(false);
    const { errorMessage } = useSelector(state => state.pokemonStore);
    
    const handleOpen = () => setOpen(true);
    

    const handleClose = () => {
        setOpen(false);
        dispatch(resetNoApiResults());
    };


    useEffect(() => {
        if (errorMessage.length > 0) {
            handleOpen();
        }
    }, [errorMessage]);

    
    return (
        
        <Modal
            open={open}
            onClose={handleClose}
            aria-labelledby="modal-modal-title"
            aria-describedby="modal-modal-description"
        >
            <Box
                textAlign='center'
                sx={style}
            >
                <Error fontSize='large' color='primary' />
                <Typography id="modal-modal-title" variant="h6" component="h2" color="primary">
                    NO HAY RESULTADOS DISPONIBLES.
                </Typography>
                <Typography id="modal-modal-description" sx={{ mt: 2 }}>
                    {errorMessage}.
                </Typography>
            </Box>
        </Modal>
        
    );
};
