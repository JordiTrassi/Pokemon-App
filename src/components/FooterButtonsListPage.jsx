import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { Grid, Button, Tooltip, IconButton } from '@mui/material';
import ArrowCircleLeftIcon from '@mui/icons-material/ArrowCircleLeft';

import { getAllPokemons, startLoadingAllPokemons, previousPage, nextPage } from '../store';


export const FooterButtonsListPage = () => {
  
    const dispatch = useDispatch();
    // const navigate = useNavigate();
  
    const { isLoading, renderPage, page } = useSelector(state => state.pokemonStore);

    const getMorePokemons = () => {
        dispatch(startLoadingAllPokemons());
        dispatch(getAllPokemons(page));
    }

    const goPreviousPage = () => {
        const previuosRenderedPage = renderPage - 1;
        dispatch(previousPage(previuosRenderedPage))
    }

    const goNextPage = () => {
        (renderPage === page)
            ? getMorePokemons()
            : dispatch(nextPage());
    }

    return (
        <Grid
            container
            justifyContent='center'
        >
            <Grid item>
                <IconButton
                    disabled={isLoading || renderPage === 1}
                    onClick={goPreviousPage}
                    sx={{
                        color: '#d50000',
                        ':hover': { opacity: 0.5 },
                    }}
                >
                    <Tooltip
                        title="back"
                        arrow
                    >
                    <ArrowCircleLeftIcon sx={{ fontSize: 45 }} />
                    </Tooltip>
                </IconButton>
            </Grid>

            <Tooltip
                title="more results"
                arrow
            >
                <Button
                    variant="contained"
                    disabled={isLoading}
                    onClick={goNextPage}
                    sx={{ p: 2, mb: '30px', borderRadius: 50 }}
                    >
                    Next Page
                </Button>           
            </Tooltip>
        </Grid>
        
    )
};

