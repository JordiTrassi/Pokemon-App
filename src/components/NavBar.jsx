import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';

import { AppBar, Box, Toolbar, Typography, InputBase, Link, Tooltip, FormControl } from '@mui/material';
import { styled, alpha } from '@mui/material/styles';
import SearchIcon from '@mui/icons-material/Search';

import { verifyInputValue } from '../helpers/verifyInputValue';
import { getPokemonByName, startLoadingPokemonByName } from '../store';


const Search = styled('div')(({ theme }) => ({
  position: 'relative',
  borderRadius: theme.shape.borderRadius,
  backgroundColor: alpha(theme.palette.common.white, 0.15),
  '&:hover': {
    backgroundColor: alpha(theme.palette.common.white, 0.25),
  },
  marginLeft: 0,
  width: '50%',
  [theme.breakpoints.up('sm')]: {
    marginLeft: theme.spacing(1),
    width: 'auto',
  },
}));

const SearchIconWrapper = styled('div')(({ theme }) => ({
  padding: theme.spacing(0, 2),
  height: '100%',
  position: 'absolute',
  pointerEvents: 'none',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
}));

const StyledInputBase = styled(InputBase)(({ theme }) => ({
  color: 'inherit',
  '& .MuiInputBase-input': {
    padding: theme.spacing(1, 1, 1, 0),
    paddingLeft: `calc(1em + ${theme.spacing(4)})`,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: '20ch',
      '&:focus': {
        width: '30ch',
      },
    },
  },
}));


export const NavBar = () => {

  const dispatch = useDispatch();
  const [inputValue, setInputValue] = useState('');
  
  const onInputChange = ({target}) => {
    setInputValue(target.value);
  }

  const onSubmit = async () => {
    const verifiedInputValue = await verifyInputValue(inputValue);

    dispatch(startLoadingPokemonByName({ verifiedInputValue }));
    dispatch(getPokemonByName({ verifiedInputValue }));
  }

  useEffect(() => {
    const listener = event => {
      if (event.code === "Enter" || event.code === "NumpadEnter") {
        event.preventDefault();
        onSubmit();
      }
    };
    document.addEventListener("keydown", listener);
    return () => {
      document.removeEventListener("keydown", listener);
    };
  }, [inputValue]);

  return (
    
    <Box sx={{ flexGrow: 1 }}>    
      <AppBar
        position="static"
        sx={{backgroundColor: 'primary'}}
      >
        <Toolbar>          
          <Typography
                variant="h5"
                noWrap
                component="div"
                sx={{ flexGrow: 1, display: { xs: '2', sm: 'block' } }}
          >
            <Tooltip
              title="home"
              arrow
              placement="bottom"  
            >
              <Link
                color="inherit"
                underline="none" 
                href="/"
              >
                <img
                  src={`https://seeklogo.com/images/P/pokeball-logo-DC23868CA1-seeklogo.com.png`}
                  alt="homeBtn"
                  style={{
                    height: 50,
                    width: 'auto',
                    padding: 2,
                    marginTop: 5
                  }}
                />
                  
              </Link>
            </Tooltip>
          </Typography>
          <Search>
            <SearchIconWrapper>
                <SearchIcon />
            </SearchIconWrapper>
            <FormControl
              onSubmit={onSubmit}
            >
              <StyledInputBase
                type="text"
                placeholder="Search Pokemon"
                value={inputValue}
                onChange={onInputChange}
              />
            </FormControl>
          </Search>
        </Toolbar>
      </AppBar>
    </Box>
  );
}