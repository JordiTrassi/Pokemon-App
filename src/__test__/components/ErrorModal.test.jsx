import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ErrorModal } from '../../components/ErrorModal';
import { store } from '../../store';

describe('Test the <ErrorModal /> component', () => {
  
    // const open = true;
    // const errorMessage = "There are't results";

    test('Should match with the snapshot ', () => {
        
        const { container } = render(
            <Provider store={store}>
                <ErrorModal />
            </Provider>
        );
        screen.debug()

        expect(container).toMatchSnapshot();

    });

    
});
