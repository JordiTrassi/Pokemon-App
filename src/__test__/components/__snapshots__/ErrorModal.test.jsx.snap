// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the <ErrorModal /> component Should match with the snapshot  1`] = `<div />`;
